R/lazyframe-opt_flags.R

```diff
@@ -0,0 +1,138 @@
+# nolint start: object_length_linter
+
+# The env for storing QueryOptFlags methods
+polars_query_opt_flags__methods <- new.env(parent = emptyenv())
+
+QueryOptFlags <- new_class(
```

Comment:
Can you add documentation (including examples) for this, so that we can directly refer to it when we start using `pl$QueryOptFlags()` in other places?

Reply:
I don't understand why it's marked as internal. I thought the usecase would be something like this:

```r
lf$collect(optimizations = pl$QueryOptFlags())
```

since most of the individual optimization parameters are deprecated in Python Polars: https://docs.pola.rs/api/python/dev/reference/lazyframe/api/polars.LazyFrame.collect.html. In which case this function needs to be exported.

That's also why I had in mind to have examples like this:

```r
opts <- pl$QueryOptFlags()
opts

opts <- pl$QueryOptFlags(predicate_pushdown = FALSE)
opts
```

Did you want to keep this function internal only?

Reply:
Sorry, forget about it, didn't notice there were already examples